app/controllers/user.js

```diff
@@ -44,6 +44,46 @@ exports.create = (req, res, next) => {
   
 };
 
+exports.signin = (req, res, next) => {
+
+  let input = emptyToNull(req.body);
+
+  if(!input.email || !input.password){
+    return res.status(400).send('Both email and password are required to continue');
+  }
+  if (req.session && req.session.user && req.session.user.email === input.email){
+    return res.status(200).send('You are already logged in!');
+    
+  }
+
+  User.findOne({
+    where: {
+      email: input.email
+    }
+  }).then(result => {
+    
+    if(!result || !result.email){
+      logger.info(`Login attempt with invalid email: "${input.email}"`);
+      return res.status(404).send('The provided email does not exist in our database!');
```

Comment:
In any other scenario the different status codes are great, but in particular in the login endpoint its a good security practice to always return 401 and some generic error. So you don't provide a possible hacker info about which emails exist and which doesn't.